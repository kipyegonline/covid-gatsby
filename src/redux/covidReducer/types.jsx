export const C = {
    ADD_COVID: "ADD_COVID",
    SORT_COVID: "SORT_COVID",
    ADD_TIMESET: "ADD_TIMESET",
    ADD_COUNTRY: "ADD_COUNTRY",
    REMOVE_COUNTRY: "REMOVE_COUNTRY",
    LOADING_MESSAGE: "LOADING_MESSAGE",
    SET_MONTH: "SET_MONTH",
    BTN_CLICKED: "BTN_CLICKED",
    CLICKED_BTN: "CLICKED_BTN"
}