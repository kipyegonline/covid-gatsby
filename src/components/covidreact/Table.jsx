import React from "react"
import { formatNums, formatNumLocale } from "../helpers"
import styled from "styled-components"

const TableEl = styled.table.attrs(props => ({
  className: props.classList,
}))``
const Table = ({ tableData, region }) => (
  <div className="table-responsive">
    <p className="text-center font-weight-bold">COVID19 infections {region}</p>
    <hr />
    <TableEl classlist="table table-bordered table-hover">
      <thead>
        <tr>
          <th>#</th>
          <th>Country</th>
          <th>Total Cases</th>
          <th>Total Deaths</th>
          <th>Recovered</th>
        </tr>
      </thead>
      <tbody>
        {tableData.map((item, i) => (
          <TableContent key={i} index={i} {...item} />
        ))}
      </tbody>
    </TableEl>
  </div>
)
const TableContent = ({
  Country,
  index,
  flag,
  TotalConfirmed,
  NewConfirmed,
  TotalDeaths,
  TotalRecovered,
}) => (
  <tr>
    <td>{index + 1}</td>
    <td>
      {Country} <img src={flag} width={20} height={10} />
    </td>
    <td className="text-info">{formatNumLocale(TotalConfirmed)}</td>
    <td className="text-danger">{formatNumLocale(TotalDeaths)}</td>
    <td className="text-success">{formatNumLocale(TotalRecovered)}</td>
  </tr>
)
export default Table
