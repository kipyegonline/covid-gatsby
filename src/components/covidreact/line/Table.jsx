import React, { useState } from "react";
import { formatNums, formatNumLocale, paginationDevice } from "../../helpers";
import styled from "styled-components";

const TableEl = styled.table.attrs((props) => ({
  className: props.classList,
}))``;
const Button = styled.button.attrs((props) => ({
  className: props.classlist,
  onClick: props.onClick,
}))`
  outline: none;
  max-width: 200px;
  border: none;
  margin-top: 0.5rem;

  &:active {
    outline: none;
    border: none;
  }
`;

function Table({ tableData, remove, add }) {
  const perPage = 10;
  const [pageNos, setpageNos] = useState(Math.ceil(tableData.length / perPage));
  const [current, setCurrent] = useState(0);

  const getValue = (click) => {
    if (click <= 0) {
      return setCurrent(pageNos - 1);
    }
    if (click >= pageNos) {
      return setCurrent(0);
    }
    setCurrent(click);
  };

  return (
    <div className="table-responsive">
      <Pointer
        text={"<<"}
        classlist="float-left"
        current={current - 1}
        setValue={getValue}
      />
      <Pointer
        text={">>"}
        current={current + 1}
        setValue={getValue}
        classlist="float-right"
      />
      <p className="text-center mt-4">
        Page {current + 1} of {pageNos}
      </p>
      <TableEl classlist="table table-bordered table-hover">
        <thead>
          <tr>
            <th>#</th>
            <th>Country</th>
            <th>Confirmed</th>
            <th>Deaths</th>
            <th>Visualize</th>
          </tr>
        </thead>
        <tbody>
          {tableData.slice(current * 10, current * 10 + 10).map((item, i) => (
            <TableContent
              key={i}
              index={current * 10 + i}
              data={item}
              add={add}
              remove={remove}
            />
          ))}
        </tbody>
      </TableEl>
      <Pointer
        text={"<<"}
        classlist="float-left"
        current={current - 1}
        setValue={getValue}
      />

      <Pointer
        text={">>"}
        current={current + 1}
        setValue={getValue}
        classlist="float-right"
      />
      <p className="text-center my-4">
        Page {current + 1} of {pageNos}
      </p>
    </div>
  );
}

const TableContent = ({ data, add, remove, index, flag }) => (
  <tr>
    <td>{index + 1}.</td>
    <td>
      {data.countryregion}{" "}
      <img src={data.flag} width={20} alt="flag" height={10} />
    </td>
    <td className="text-info">{formatNumLocale(data.latest)}</td>
    <td className="text-danger">
      {formatNumLocale(data.stats[data.stats.length - 1].deaths)}
    </td>
    <td>
      {!data.added ? (
        <Button
          onClick={() => add(data)}
          classlist="blue btn btn-info btn-block  darken-2"
        >
          Add{" "}
        </Button>
      ) : (
        <Button
          onClick={() => remove(data)}
          classlist=" btn btn-block white-text red darken-2"
        >
          Remove
        </Button>
      )}
    </td>
  </tr>
);
const setStyle = () => ({
  fontWeight: "bold",
  fontSize: "2rem",
  color: "grey",
  lineHeight: "1em",
  padding: "1rem",
});

const Pointer = ({ text, classlist, setValue, current }) => (
  <span
    style={setStyle()}
    onClick={() => setValue(current)}
    className={classlist}
  >
    {text}
  </span>
);
export default Table;
