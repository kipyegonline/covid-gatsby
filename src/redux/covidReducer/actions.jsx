import { C } from "./types"

export const addCovid = (payload) => ({
    type: C.ADD_COVID,
    payload
})
export const addTimeSeries = (data) => ({
    type: C.ADD_TIMESET,
    payload: transformData(data)
})


export const addCountry = (payload) => ({
    type: C.ADD_COUNTRY,
    payload
})

export const removeCountry = (payload) => ({
    type: C.REMOVE_COUNTRY,
    payload
})

export const loadingMessage = (payload) => ({
    type: C.LOADING_MESSAGE,
    payload
})
export const setMonth = (payload) => ({
    type: C.SET_MONTH,
    payload
})
export const sortCovid = (sort, cwd) => ({
    type: C.SORT_COVID, sort, sorted: cwd
})
export const sendBtn = (payload) => ({
    type: C.BTN_CLICKED,
    payload
})




const fetchData = async (url) => {
    try {

        const res = await fetch(url);

        if (res.status) {
            const data = await res.json();
            return data.Countries;
        }

    }
    catch (error) {
        console.log('fetch err data:', error.message)
    }


}

const fetchContinent = async (region) => {
    try {

        let global = "https://restcountries.eu/rest/v2/all"

        let url = region === "global" ? global : `https://restcountries.eu/rest/v2/region/${region}`

        const res = await fetch(url);
        const data = await res.json();

        let dataset = data.map(item => {
            if (item.name === "United States of America") {
                item.name = "US"
            }
            return {
                name: item.name,
                flag: item.flag
            }
        })
        return dataset;
    }
    catch (error) {
        console.log('fetch err', error.message)
    }

}



export function getNations(data, continent, countryKey, callback) {
    let dataholder = []

    if (data) {

        dataholder = data.filter(item => continent.find(afr => {

            if (item[countryKey] === "United States of America") {
                item[countryKey] = "US";
            }
            if (afr["name"] === item[countryKey]) {
                item.flag = afr.flag;
                return item
            }

        })





        )


        callback(dataholder)


    }
}

export async function covid19(callback, ur1, region, tracker) {

    try {
        tracker({ msg: "loading " + region, spin: true })

        const [covid, africa] = await Promise.all([fetchData(ur1), fetchContinent(region)])
        if (covid === undefined || africa === undefined) {
            tracker({
                msg: `Error loading ${region}..try again`, spin: true
            })
            setTimeout(() => tracker({ msg: "", spin: false }), 2000)
        } else {
            getNations(covid, africa, "Country", callback)
            tracker({ msg: "", spin: false })
        }

    }
    catch{
        console.log("promise errrr")
    }

}


///get time series of cases


const fetchEntry = ur1 => fetch(ur1).then(res => res.json()).then(data => data).catch(err => console.error(err.message));

export async function covid19Series(callback, ur1, region, tracker) {

    try {

        tracker({ msg: "loading data", spin: true });

        const [covidData, continent] = await Promise.all([fetchEntry(ur1), fetchContinent(region)])
        // console.log('seriesssss', covidData, continent)
        if (covidData === undefined || continent === undefined) {

            tracker({ msg: "Error loading..try again", spin: false })
            setTimeout(() => tracker({ msg: "", spin: false }), 2000)
        } else {
            tracker({ msg: "", spin: false })
            getNations(covidData, continent, "countryregion", callback, tracker)
        }

    }
    catch (error) {
        console.log("Line Error: ", error.message)
        tracker({ msg: error.message, spin: false })

    }


}


export const transformData = (dataset) => {
    if (dataset) {
        let counter = 0;


        return dataset.map(data => {
            let timeData = []
            const bigO = data.timeseries
            for (let ob in bigO) {

                const confirmed = bigO[ob].confirmed;
                const deaths = bigO[ob].deaths;
                const recovered = bigO[ob].recovered;
                const stats = {
                    date: ob,
                    confirmed,
                    deaths,
                    recovered
                }

                timeData = [...timeData, stats]
            }
            data.latestupdate = new Date(data.lastupdate.split("T")[0])
            let count = 0;
            data.stats = timeData.map(time => {
                return {
                    ...time,
                    date: new Date(time.date),
                    id: count++
                }


            })
            data.latest = timeData[timeData.length - 1].confirmed;
            data.deaths = timeData[timeData.length - 1].deaths;
            data.recovered = timeData[timeData.length - 1].recovered;
            data.date = timeData[timeData.length - 1].date;
            data.id = counter++;

            return data;
        })
            .sort((a, b) => b.latest - a.latest)
    }

}
let count = 0;
export const filterByMonth = (dataset, month) => {
    let holder = [];

    const res = dataset.map(d => {
        return d.filter(item => {
            if (item.date.getMonth() >= month) {

                return item
            }
        })
    })

    return res;
}

