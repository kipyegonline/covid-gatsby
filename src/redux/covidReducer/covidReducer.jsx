import { C } from "./types";
import { filterByMonth } from "./actions";

const continents = [
  "Global",
  "Africa",
  "Europe",
  "Asia",
  "Americas",
  "Oceania",
];
const months = [
  { id: 0, month: "January" },
  { id: 1, month: "February" },
  { id: 2, month: "March" },
  { id: 3, month: "April" },
  { id: 4, month: "May" },
  { id: 5, month: "June" },
  { id: 6, month: "July" },
  { id: 7, month: "August" },
];
const btns = [
  { continent: "Africa", val: "africa", id: 0, selected: true },

  { continent: "Europe", val: "europe", id: 1, selected: false },

  { continent: "Asia", val: "asia", id: 2, selected: false },
  { continent: "Americas", val: "americas", id: 3, selected: false },
  { continent: "Oceania", val: "oceania", id: 4, selected: false },
];
const initState = {
  covidDataset: [],
  cwd: [],
  sort: true,
  timeDataset: [],
  eachDay: [],
  month: 2,
  selectedCountry: [],
  btns,
  message: "",
  continents,
  spinner: false,
  months,
  clickedBtn: "africa",
};
function covidReducer(state = initState, action) {
  switch (action.type) {
    case C.ADD_COVID:
      return {
        ...state,
        covidDataset: action.payload,
        cwd: action.payload,
        sort: true,
      };
      break;

    //timeseries array from the api
    case C.ADD_TIMESET:
      // add the first four items in the array to the graph
      const newState = [...action.payload].slice(0, 5).map((d) => d.stats);

      return {
        ...state,
        timeDataset: update5(action.payload),
        selectedCountry: [...action.payload].slice(0, 5),
        eachDay:
          state.month > 0 ? filterByMonth(newState, state.month) : newState,
      };
    //added country
    case C.ADD_COUNTRY:
      if (state.selectedCountry.length >= 10) {
        alert("You can only add upto 10 countries.");
        return state;
      }
      const inc = action.payload.id;
      const checker = state.selectedCountry.find((d) => d.id === inc);
      if (checker === undefined) {
        //add country to assist when removing from store
        action.payload.stats[action.payload.stats.length - 1].country =
          action.payload.countryregion;
        const newState = [...state.eachDay, action.payload.stats];

        return {
          ...state,
          timeDataset: [...state.timeDataset].map((data) =>
            data.id === action.payload.id ? { ...data, added: true } : data
          ),
          selectedCountry: [...state.selectedCountry, action.payload],
          eachDay:
            state.month > 0 ? filterByMonth(newState, state.month) : newState,
        };
      } else {
        return state;
      }

    //remove
    case C.REMOVE_COUNTRY:
      if (state.selectedCountry.length <= 1) {
        /* eslint-disable no-alert */
        alert("choose a default country");
        return state;
      }

      return {
        ...state,
        timeDataset: [...state.timeDataset].map((data) =>
          data.id === action.payload.id ? { ...data, added: false } : data
        ),
        selectedCountry: state.selectedCountry.filter(
          (country) => country.id !== action.payload.id
        ),
        eachDay: state.eachDay.filter(
          (d) =>
            d[d.length - 1].country !==
            action.payload.stats[action.payload.stats.length - 1].country
        ),
      };

    //for showing errors during ajax call
    case C.LOADING_MESSAGE:
      return {
        ...state,
        message: action.payload.msg,
        spinner: action.payload.spin,
      };
    case C.SET_MONTH:
      const c = state.timeDataset.slice();
      const res = c
        .filter((data) => data.added === true)
        .map((data) => data.stats);

      return {
        ...state,
        month: action.payload,
        eachDay: filterByMonth(res, action.payload),
      };

    case C.BTN_CLICKED:
      return {
        ...state,
        clickedBtn: state.btns.find((btn) => btn.id === action.payload).val,
        btns: state.btns.map((btn) =>
          btn.id === action.payload
            ? { ...btn, selected: true }
            : { ...btn, selected: false }
        ),
      };

    case C.SORT_COVID:
      return {
        ...state,
        sort: !state.sort,
        covidDataset: state.sort ? action.sorted : state.cwd,
      };
    default:
      return state;
  }
}
export default covidReducer;
//selected country reducer
const covid = ({}, action) => {
  if (action.type === C.ADD_COUNTRY) {
    return { ...action.payload };
  }
};

//timeseries reducer
const covidTimeseries = ([], action) => {
  if (action.payload === C.ADD_TIMESET) {
    return action.payload;
  }
};

const update5 = (data) => {
  let res = [];
  const sel = data.slice(0, 5);
  const dd = data.map((d) => {
    let seen = sel.find((s) => s.id === d.id);
    if (seen) {
      seen.stats[d.stats.length - 1].country = d.countryregion;
      return { ...seen, added: true };
    } else {
      d.stats[d.stats.length - 1].country = d.countryregion;
      return { ...d, added: false };
    }
  });

  return dd;
};
