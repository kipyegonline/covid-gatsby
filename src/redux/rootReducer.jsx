import { combineReducers } from "redux"
import covidReducer from "./covidReducer/covidReducer"
import keReducer from "./kenyaReducer/keReducer"
export default combineReducers({ covid: covidReducer, ke: keReducer })