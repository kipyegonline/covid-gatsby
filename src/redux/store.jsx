import React from "react"
import { createStore } from "redux"
import { Provider } from "react-redux"
import rootReducer from "./rootReducer"

const devtools = () => {
  if (globalThis.window) {
    return window.__REDUX_DEVTOOLS_EXTENSION__
      ? window.__REDUX_DEVTOOLS_EXTENSION__()
      : f => f
  }
}
export const store = createStore(rootReducer, devtools())

const ReduxWrapper = ({ element }) => (
  <Provider store={store}>{element}</Provider>
)
export default ReduxWrapper
