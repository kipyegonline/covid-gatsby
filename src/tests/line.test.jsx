import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import { createStore } from "redux";
import { Provider } from "react-redux";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Table from "../components/covidreact/line/Table";
import Line from "../components/covidreact/line/line";
import covidReducer from "../redux/rootReducer";

Enzyme.configure({ adapter: new Adapter() });
const store = createStore(covidReducer);
const ReduxWrapper = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

const tableData = [
  {
    countryregion: "Kenya",
    added: false,
    flag: "kenya.jpeg",
    stats: [{ deaths: 314, recovered: 12, confirmed: 3647 }],
  },
  {
    countryregion: "Tanzania",
    added: true,
    flag: "tanzania.jpeg",
    stats: [{ deaths: 650, recovered: 234, confirmed: 7648 }],
  },
];
store.dispatch({
  type: "ADD_KE",
  payload: [
    {
      countryregion: "Kenya",
      id: 1,
      stats: [
        {
          deaths: 50,
          recovered: 10,
          confirmed: 20,
          date: new Date("2020/05/01"),
          id: 0,
        },
        {
          deaths: 50,
          recovered: 10,
          confirmed: 20,
          date: new Date("2020/05/02"),
          id: 1,
        },
        {
          deaths: 250,
          recovered: 100,
          confirmed: 60,
          date: new Date("2020/05/03"),
          id: 2,
        },
        {
          deaths: 500,
          recovered: 410,
          confirmed: 60,
          date: new Date("2020/05/04"),
          id: 3,
        },
        {
          deaths: 650,
          recovered: 300,
          confirmed: 800,
          date: new Date("2020/05/05"),
          id: 4,
        },
      ],
    },
  ],
});

describe("<Line/> component", () => {
  const _getTools = jest.fn(),
    _getOp = jest.fn();

  const wrapper = mount(
    <ReduxWrapper>
      <Line
        data={tableData}
        getTools={_getTools}
        getOp={_getOp}
        month={"January"}
      />
    </ReduxWrapper>
  );
  console.log(wrapper.debug());
});
describe("<Table/> component", () => {
  const _remove = jest.fn();
  const _add = jest.fn();

  const wrapper = mount(
    <Table add={_add} tableData={tableData} remove={_remove} />
  );

  describe("UI layout", () => {
    test("matches snapshot", () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
    test("presence of 2 table items", () => {
      expect(wrapper.find("tr")).toHaveLength(3);
    });
    test("first item", () => {
      const first = wrapper.find("tr").at(1);
      expect(first.find("img").props().src).toBe("kenya.jpeg");
      expect(first.find("button")).toHaveLength(1);
      expect(
        first.find("button").hasClass("blue btn btn-info btn-block  darken-2")
      ).toBeTruthy();
      first.find("button").simulate("click");
      expect(_add).toHaveBeenCalled();
      expect(_add).toHaveBeenCalledWith(tableData[0]);
      expect(
        first.find("button").hasClass("blue btn btn-info btn-block  darken-2")
      ).toBeTruthy();
    });
    test("second item", () => {
      const first = wrapper.find("tr").at(2);
      expect(first.find("img").props().src).toBe("tanzania.jpeg");
      expect(first.find("button").text()).toBe("Remove");
      expect(first.find("button")).toHaveLength(1);
      expect(
        first.find("button").hasClass("btn btn-block white-text red darken-2")
      ).toBeTruthy();
      first.find("button").simulate("click");
      expect(_remove).toHaveBeenCalled();
      expect(_remove).toHaveBeenCalledWith(tableData[1]);
      expect(
        first.find("button").hasClass("blue btn btn-info btn-block  darken-2")
      ).toBeFalsy();
    });
  });
});
