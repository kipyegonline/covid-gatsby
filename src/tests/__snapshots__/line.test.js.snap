// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table/> component UI layout matches snapshot 1`] = `
<Table
  add={[MockFunction]}
  remove={[MockFunction]}
  tableData={
    Array [
      Object {
        "added": false,
        "countryregion": "Kenya",
        "flag": "kenya.jpeg",
        "stats": Array [
          Object {
            "confirmed": 3647,
            "deaths": 314,
            "recovered": 12,
          },
        ],
      },
      Object {
        "added": true,
        "countryregion": "Tanzania",
        "flag": "tanzania.jpeg",
        "stats": Array [
          Object {
            "confirmed": 7648,
            "deaths": 650,
            "recovered": 234,
          },
        ],
      },
    ]
  }
>
  <div
    className="table-responsive"
  >
    <Pointer
      classlist="float-left"
      current={-1}
      setValue={[Function]}
      text="<<"
    >
      <span
        className="float-left"
        onClick={[Function]}
        style={
          Object {
            "color": "grey",
            "fontSize": "2rem",
            "fontWeight": "bold",
            "lineHeight": "1em",
            "padding": "1rem",
          }
        }
      >
        &lt;&lt;
      </span>
    </Pointer>
    <Pointer
      classlist="float-right"
      current={1}
      setValue={[Function]}
      text=">>"
    >
      <span
        className="float-right"
        onClick={[Function]}
        style={
          Object {
            "color": "grey",
            "fontSize": "2rem",
            "fontWeight": "bold",
            "lineHeight": "1em",
            "padding": "1rem",
          }
        }
      >
        &gt;&gt;
      </span>
    </Pointer>
    <p
      className="text-center mt-4"
    >
      Page 
      1
       of 
      1
    </p>
    <styled.table
      classlist="table table-bordered table-hover"
    >
      <table
        className="sc-AxjAm StDqN"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Country
            </th>
            <th>
              Confirmed
            </th>
            <th>
              Deaths
            </th>
            <th>
              Visualize
            </th>
          </tr>
        </thead>
        <tbody>
          <TableContent
            add={[MockFunction]}
            data={
              Object {
                "added": false,
                "countryregion": "Kenya",
                "flag": "kenya.jpeg",
                "stats": Array [
                  Object {
                    "confirmed": 3647,
                    "deaths": 314,
                    "recovered": 12,
                  },
                ],
              }
            }
            index={0}
            key="0"
            remove={[MockFunction]}
          >
            <tr>
              <td>
                1
                .
              </td>
              <td>
                Kenya
                 
                <img
                  alt="flag"
                  height={10}
                  src="kenya.jpeg"
                  width={20}
                />
              </td>
              <td
                className="text-info"
              >
                0
              </td>
              <td
                className="text-danger"
              >
                314
              </td>
              <td>
                <styled.button
                  classlist="blue btn btn-info btn-block  darken-2"
                  onClick={[Function]}
                >
                  <button
                    className="sc-AxirZ hnqJkY blue btn btn-info btn-block  darken-2"
                    onClick={[Function]}
                  >
                    Add
                     
                  </button>
                </styled.button>
              </td>
            </tr>
          </TableContent>
          <TableContent
            add={[MockFunction]}
            data={
              Object {
                "added": true,
                "countryregion": "Tanzania",
                "flag": "tanzania.jpeg",
                "stats": Array [
                  Object {
                    "confirmed": 7648,
                    "deaths": 650,
                    "recovered": 234,
                  },
                ],
              }
            }
            index={1}
            key="1"
            remove={[MockFunction]}
          >
            <tr>
              <td>
                2
                .
              </td>
              <td>
                Tanzania
                 
                <img
                  alt="flag"
                  height={10}
                  src="tanzania.jpeg"
                  width={20}
                />
              </td>
              <td
                className="text-info"
              >
                0
              </td>
              <td
                className="text-danger"
              >
                650
              </td>
              <td>
                <styled.button
                  classlist=" btn btn-block white-text red darken-2"
                  onClick={[Function]}
                >
                  <button
                    className="sc-AxirZ hnqJkY  btn btn-block white-text red darken-2"
                    onClick={[Function]}
                  >
                    Remove
                  </button>
                </styled.button>
              </td>
            </tr>
          </TableContent>
        </tbody>
      </table>
    </styled.table>
    <Pointer
      classlist="float-left"
      current={-1}
      setValue={[Function]}
      text="<<"
    >
      <span
        className="float-left"
        onClick={[Function]}
        style={
          Object {
            "color": "grey",
            "fontSize": "2rem",
            "fontWeight": "bold",
            "lineHeight": "1em",
            "padding": "1rem",
          }
        }
      >
        &lt;&lt;
      </span>
    </Pointer>
    <Pointer
      classlist="float-right"
      current={1}
      setValue={[Function]}
      text=">>"
    >
      <span
        className="float-right"
        onClick={[Function]}
        style={
          Object {
            "color": "grey",
            "fontSize": "2rem",
            "fontWeight": "bold",
            "lineHeight": "1em",
            "padding": "1rem",
          }
        }
      >
        &gt;&gt;
      </span>
    </Pointer>
    <p
      className="text-center my-4"
    >
      Page 
      1
       of 
      1
    </p>
  </div>
</Table>
`;
